.github/scripts/findUnusedStyles.ts

```diff
@@ -0,0 +1,462 @@
+                        shouldCapture = true;
+                    }
+
+                    if (shouldCapture) {
+                        const {line, character} = sourceFile.getLineAndCharacterOfPosition(node.getStart());
+                        this.styleDefinitions.set(key, {
+                            key,
+                            file,
+                            line: line + 1,
+                            column: character + 1,
+                        });
+                    }
+                } catch (error) {
+                    console.warn(`Warning: Could not get position for style '${key}' in ${file}`);
+                }
+            }
+        }
+
+        // Also look for variable declarations that might be styles
+        if (ts.isVariableDeclaration(node) && ts.isIdentifier(node.name)) {
+            const key = node.name.text;
+            if (this.isStyleKey(key)) {
+                try {
+                    const {line, character} = sourceFile.getLineAndCharacterOfPosition(node.getStart());
+                    this.styleDefinitions.set(key, {
+                        key,
+                        file,
+                        line: line + 1,
+                        column: character + 1,
+                    });
+                } catch (error) {
+                    console.warn(`Warning: Could not get position for style '${key}' in ${file}`);
+                }
+            }
+        }
+    }
+
+    private isStyleKey(key: string): boolean {
+        // Skip certain patterns that are likely not style keys
+        if (STYLE_KEY_SKIP_PATTERNS.includes(key)) {
+            return false;
+        }
+
+        // Must be a valid identifier
+        if (!VALID_IDENTIFIER_PATTERN.test(key)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private loadAllFileContents(): void {
+        // Load all source files EXCEPT style definition files
+        const allFilesPaths = `src/**/${STYLE_FILE_EXTENSIONS}`;
+        const allFiles = globSync(allFilesPaths, {
+            cwd: this.rootDir,
+        });
+
+        // Filter out style definition files, BUT keep utils/generators/themes for usage checking
+        const sourceFiles = allFiles.filter((file) => {
+            // Keep utils/generators/themes to check for main style usage
+            if (EXCLUDED_STYLE_DIRECTORIES.some((dir) => file.includes(dir))) {
+                return true;
+            }
+            // Exclude other style definition files (like index.ts itself)
+            return !file.includes('src/styles/');
+        });
+
+        console.log(`Loading ${sourceFiles.length} source files (including utils/generators/themes for usage checking)...`);
+
+        for (const file of sourceFiles) {
+            const fullPath = path.join(this.rootDir, file);
+            try {
+                // Check if it's actually a file (not a directory)
+                const stat = fs.lstatSync(fullPath);
```

Comment:
NAB but you're doing `sync` operations in a loop here. You could use async operations then do an `await Promise.all` after the loop. Not sure it really matters but that's how I would implement it.

Reply:
no problem